File: bench/ruby/sort.rb
n = (ENV["BENCH_N"] || "750000").to_i
mod = 1000000007

x = 123456789
data = Array.new(n) do
  x = (x * 1103515245 + 12345) & 0x7FFFFFFF
  x
end

data.sort!

h = 0
data.each { |v| h = (h * 31 + v) % mod }

puts h

File: bench/node/sort.js
const N = parseInt(process.env.BENCH_N || "375000", 10);
const MOD = 1000000007;
let x = 123456789;
const data = new Array(N);
for (let i = 0; i < N; i++) {
  x = (Math.imul(x, 1103515245) + 12345) & 0x7FFFFFFF;
  data[i] = x;
}
data.sort((a, b) => a - b);
let h = 0;
for (const v of data) h = (h * 31 + v) % MOD;
console.log(h);
